connector/connect/common/src/main/protobuf/spark/connect/relations.proto

```diff
@@ -211,6 +211,11 @@ message SetOperation {
   // Only UNION supports this option.
   optional bool by_name = 5;
 
+  // (Optional) If to perform the Set operation and allow missing columns.
```

Comment:
cc @amaliujia 

it seems that `is_all` is only used in `Intersect` and `Except`,
`by_name` and `allow_missing_columns` are only used in `Union`.

Maybe we should divide the `SetOperation` into three separate protos?

Reply:
happen to find that there is an abstract class `SetOperation` in scala, but `Union` is not its subclass

Reply:
This was designed intentionally. 

Basically we could also extend `by_name` and `allow_missing_columns` to `intersect` and `except` (even though we do not have it now) so we believed that they all belong to a SetOperation which includes UNION, INTERSECT, EXCEPT and sharing the same set of parameters.

Reply:
> Basically we could also extend by_name to intersect and except (even though we do not have it now) 

make sense, thanks for clarification